### PR TITLE
Extend inline boundaries to other elements

### PR DESCRIPTION
**Pull request by BoardJames**
## Description
Inline boundaries are extended to code, b, i, strong, em, del, ins, sub, sup
This means that an extra cursor position exists on the boundaries of these tags and they highlight when the cursor is within them.

## How Has This Been Tested?
This has been manually tested in Firefox and Chrome.
It is a core part of TinyMCE that we're enabling through configuration so I do not believe that it requires automated tests in Gutenberg.

## Types of changes
This is a new feature.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style.
- [x] My code follows has proper inline documentation.

### CODE REVIEW

**Comment by androb** [comment]
@EphoxJames , you will want to check that the styling in Gutenberg covers these other inline elements as well.

Something like:
content_style: [
    '.mce-content-body a[data-mce-selected], .mce-content-body code[data-mce-selected], .mce-content-body b[data-mce-selected], .mce-content-body strong[data-mce-selected], .mce-content-body i[data-mce-selected], .mce-content-body em[data-mce-selected], .mce-content-body ins[data-mce-selected], .mce-content-body del[data-mce-selected] {background-color:#e5f5fa}']

**Comment by EphoxJames** [comment]
@androb done.

File: blocks/editable/style.scss [hunk 1]
```diff
@@ -47,6 +47,18 @@
 		background: $light-gray-400;
 	}
 
+	&:focus {
+		b, i, strong, em, del {
```
Comment (androb) [inline blocks/editable/style.scss]: I believe that ins was wanted as well

**Comment by jasmussen** [comment]
Very nice work, fast work too! Love it, thank you.

As Andrew suggested, `ins` would be nice too.

Also, can we make this work in the Classic Text block too? And while we're at it, does it makes ense to use the same line-height for the Classic Text block as we use in the Paragraph block? Screenshot:

screen shot 2017-10-10 at 08 59 09

Here's some copy pasta for testing:

```
<p><b>Loem</b> ipsum <strong>dolor</strong> sit <i>amet</i>, ferri <em>vidisse</em> nam eu, <del>ad</del> nec <ins>copiosae</ins> mnesarchum <code>vituperatoribus</code>. Te <a href="#">brute</a> dicunt sea, ut vis omnium menandri, ut sumo aliquam has. Eum aperiam interpretaris at, sea et recusabo expetenda, omnis tibique mea no. Pri suas partem ea, ius sonet numquam offendit cu, ad simul admodum pri. Eum cu unum choro albucius.</p>

<!-- wp:core/paragraph -->
<p><b>Loem</b> ipsum <strong>dolor</strong> sit <i>amet</i>, ferri <em>vidisse</em> nam eu, <del>ad</del> nec <ins>copiosae</ins> mnesarchum <code>vituperatoribus</code>. Te <a href="#">brute</a> dicunt sea, ut vis omnium menandri, ut sumo aliquam has. Eum aperiam interpretaris at, sea et recusabo expetenda, omnis tibique mea no. Pri suas partem ea, ius sonet numquam offendit cu, ad simul admodum pri. Eum cu unum choro albucius.</p>
<!-- /wp:core/paragraph -->
```

**Comment by jasmussen** [comment]
Tracked the lineheight bug in #2957.

File: blocks/editable/style.scss (hunk 1 above)
Comment (EphoxJames) [reply blocks/editable/style.scss]: @androb support for ins added, I also added support for sub and sup.

**Comment by EphoxJames** [comment]
@jasmussen The classic text block uses the TinyMCE from Wordpress so it is a few versions behind the version which has the inline boundaries on elements other than links.

**Comment by EphoxJames** [comment]
I believe this PR is ready to merge. Anyone have objections?

**Comment by jasmussen** [comment]
Just to be evil, I threw some horrendous tag soup at it, with multiple layers of nested inline elements. I'm not even going to show you the tag soup, it was that bad.

But TinyMCE handled it beautifully 🎉 — only the outermost boundary registered, as I would argue is the best possible way to handle it. 

Yep, this is good to merge! Thanks for working on it!

**Pull request merged**